### ISSUE TITLE
zig package does not build on x86_64

### ISSUE THREAD

**Issue by roestii** [rxrbln/t2sde#182]
I tried to build zig 0.13.0 using t2 install but it failed with a lot of linking errors as the one seen below:
`error: ld.lld: undefined symbol: __isoc23_strtoul`

I tried to search for the error and it might be caused by zig trying to link against glibc 2.38. I tried to fiddle around with the cmake options from the `zig.conf` in the zig package and when I removed the `-DZIG_HOST_TARGET` and the `-DZIG_TARGET` it magically worked (as seen in the attached diff). So likely a zig issue? This really seems like a workaround and I don't have the most experience with build systems so one might help me out here. 

[zig_conf.patch](https://github.com/user-attachments/files/17075552/zig_conf.patch)

Thanks!

**Comment by cosmadragos73** [comment]
yes, symbols are affected by build command changes, especially the -D, they are later on passed as a DEFINE to be sorted out.

 will need more info

**Comment by orlovskyjavaprofi** [comment]
zig package (disable this package): change zig to version 0.16.0 zig requires llvm version 21.1.0, **other versions of llvm not compatible with zig!**  
**This zig.desc do not work, zig is breaking build package!**
var_append cmakeopt ' ' "-DCMAKE_C_COMPILER=$root/$prefix/bin/clang" 
var_append cmakeopt ' ' "-DCMAKE_CXX_COMPILER=$root/$prefix/bin/clang++"  
var_append cmakeopt ' ' "-DCMAKE_LINKER=$root/$prefix/bin/lld" 
var_append cmakeopt ' ' "-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld"

var_append cmakeopt ' ' -DCMAKE_CXX_STANDARD=17
var_append cmakeopt ' ' -DLLVM_PARALLEL_LINK_JOBS=2
var_append cmakeopt ' ' "-DZIG_TARGET_TRIPLE=x86_64-linux-gnu"
var_append cmakeopt ' ' "-DLLVM_DIR=$root/$prefix/lib/cmake/llvm"
var_append cmakeopt ' ' "-DClang_DIR=$root/$prefix/lib/cmake/clang"
var_append cmakeopt ' ' "-DLLD_DIR=$root/$prefix/lib/cmake/lld"

# 1. Force the Linker to search YOUR prefix BEFORE the system
var_append cmakeopt ' ' "-DCMAKE_LIBRARY_PATH=$root/$prefix/lib"

# 2. Hardcode the path so the binary knows where its libraries are at runtime
var_append cmakeopt ' ' "-DCMAKE_INSTALL_RPATH=$root/$prefix/lib"

# 3. Tell the linker to ignore the host's problematic libraries
var_append cmakeopt ' ' "-DCMAKE_EXE_LINKER_FLAGS=-Wl,-rpath,$root/$prefix/lib"

**Comment by rxrbln** [comment]
Zig not being able to release latest llvm compatible versions is an upstream issue. Best suggest upstream. Zig currently only builds when the version matches. Patch to patch zig to work with latest stable llvm welcome.

**Comment by orlovskyjavaprofi** [comment]
**Very important:**  zig upstream plans to drop the support for **llvm** , so currently my best advise not to build zig and in future that would make sense to remove it.

**Comment by rxrbln** [comment]
I know, but upstream says this for years and it might still take years.
